nexus/test-utils/src/resource_helpers.rs

```diff
@@ -1316,35 +1345,38 @@ impl<'a, N: NexusServer> DiskTest<'a, N> {
         }
     }
 
-    pub async fn add_zpool_with_dataset(&mut self, sled_id: SledUuid) {
-        self.add_zpool_with_dataset_ext(
+    /// Adds the zpool and datasets into the database.
+    ///
+    /// Does not inform sled agents to use these pools.
+    ///
+    /// See: [Self::propagate_datasets_to_sleds] if you want to send
+    /// this configuration to a simulated sled agent.
+    pub async fn add_zpool_with_datasets(&mut self, sled_id: SledUuid) {
+        self.add_zpool_with_datasets_ext(
             sled_id,
             PhysicalDiskUuid::new_v4(),
             ZpoolUuid::new_v4(),
-            vec![TestDataset {
-                id: DatasetUuid::new_v4(),
-                kind: DatasetKind::Crucible,
-            }],
+            vec![
+                TestDataset {
+                    id: DatasetUuid::new_v4(),
+                    kind: DatasetKind::Crucible,
+                },
+                TestDataset {
+                    id: DatasetUuid::new_v4(),
+                    kind: DatasetKind::Debug,
+                },
+            ],
             Self::DEFAULT_ZPOOL_SIZE_GIB,
         )
         .await
     }
 
-    // Propagate the dataset configuration to all Sled Agents.
-    //
-    // # Panics
-    //
-    // This function will panic if any of the Sled Agents have already
-    // applied dataset configuration.
-    //
+    /// Propagate the dataset configuration to all Sled Agents.
     // TODO: Ideally, we should do the following:
-    // 1. Also call a similar method to invoke the "omicron_physical_disks_ensure" API. Right now,
-    //    we aren't calling this at all for the simulated sled agent, which only works because
-    //    the simulated sled agent simply treats this as a stored config, rather than processing it
-    //    to actually provide a different view of storage.
-    // 2. Re-work the DiskTestBuilder API to automatically deploy the "disks + datasets" config
-    //    to sled agents exactly once. Adding new zpools / datasets after the DiskTest has been
-    //    started will need to also make a decision about re-deploying this configuration.
+    // Also call a similar method to invoke the "omicron_physical_disks_ensure" API. Right now,
+    // we aren't calling this at all for the simulated sled agent, which only works because
+    // the simulated sled agent simply treats this as a stored config, rather than processing it
+    // to actually provide a different view of storage.
```

Comment:
Is the 2nd part of the old comment no longer applicable because of the update on line 1325 above?

Reply:
No, I think this comment is still accurate -- dataset config is different from physical disk config.

This comment is now saying "we should do something similar to this for disks", even though the simulated sled agent is basically ignoring the provided disks config, for now.